processor/src/continuation_stack.rs

```diff
@@ -0,0 +1,193 @@
+use alloc::{sync::Arc, vec::Vec};
+
+use miden_core::{
+    Program,
+    mast::{MastForest, MastNodeId},
+};
+
+/// Specifies the phase of a processing step.
+///
+/// Specifically, it allows to distinguish between the start and finish phases of processing a node,
+/// which gives the processor sufficient information to execute the node correctly. Note that
+/// exactly how "start" and "end" are used depends on the node type being processed.
+#[derive(Debug, Default)]
+pub enum ProcessingStepPhase {
+    #[default]
+    Start,
+    Finish,
+}
+
+/// Represents a unit of work in the continuation stack.
+///
+/// This enum defines the different types of processing steps that can be performed on MAST nodes
+/// during program execution. Each variant represents a specific phase of node processing.
+#[derive(Debug)]
+pub enum ProcessingStep {
```

Comment:
I find it a bit confusing and verbose to call this `ProcessingStep` versus `Continuation`, particularly given that this module is notionally a stack of  continuations, not processing steps. The fact that there a couple different types of continuations doesn't really change that IMO.